flink-autoscaler/src/main/java/org/apache/flink/autoscaler/tuning/MemoryTuning.java

```diff
@@ -148,8 +149,11 @@ public static ConfigChanges tuneTaskManagerHeapMemory(
         final long flinkMemoryDiffBytes = heapDiffBytes + managedDiffBytes + networkDiffBytes;
 
         // Update total memory according to memory diffs
-        final MemorySize totalMemory =
+        MemorySize totalMemory =
                 new MemorySize(maxMemoryBySpec.getBytes() - memBudget.getRemaining());
+        totalMemory =
+                MemoryScaling.applyMemoryScaling(
+                        totalMemory, maxMemoryBySpec, context, scalingSummaries, evaluatedMetrics);
```

Comment:
The memory scaling config specifies that both managed and heap memory is scaled based on the TM scaling factor. But it seems like we only scale the heap as the managed memory size is computed before this. 

Am I missing something?

Reply:
Good catch, the scaling is heap only. Managed memory is never adjusted, unless none is used at all (then it is set to zero) or the option to maximize managed memory is turned on. I've adjusted the docs.